### PR TITLE
Display sample name instead of ID in UI

### PR DESCRIPTION
**Pull request by SaschaWillems**
## Description

Samples currently only display the ID (set in the CMakeLists.txt) in the UI. This may be cryptic, e.g.:

image

This PR fixes that and displays the actual name instead:

image

## General Checklist:

Please ensure the following points are checked:

- [x] My code follows the [coding style](https://github.com/KhronosGroup/Vulkan-Samples/tree/master/CONTRIBUTING.md#Code-Style)
- [x] I have reviewed file [licenses](https://github.com/KhronosGroup/Vulkan-Samples/tree/master/CONTRIBUTING.md#Copyright-Notice-and-License-Template)
- [x] I have updated the header Copyright to reflect the current year (CI build will fail if Copyright is out of date)
- [x] I have commented any added functions (in line with Doxygen)
- [x] I have commented any code that could be hard to understand
- [x] My changes do not add any new compiler warnings
- [x] My changes do not add any new validation layer errors or warnings
- [x] I have used existing framework/helper functions where possible
- [x] My changes build on Windows, Linux, macOS and Android. Otherwise I have [documented any exceptions](https://github.com/KhronosGroup/Vulkan-Samples/tree/master/CONTRIBUTING.md#General-Requirements)
- [x] My changes do not add any regressions
- [x] I have tested every sample to ensure everything runs correctly
- [x] This PR describes the scope and expected impact of the changes I am making

### CODE REVIEW

**Comment by asuessenbach** [comment]
I like this approach.
But wouldn't it be cleaner to remove the name from SampleInfo, then?

**Review by gpx1000 — APPROVED** [review]

File: framework/platform/platform.cpp [hunk 1]
```diff
@@ -369,7 +369,7 @@ bool Platform::start_app()
 
 	active_app = requested_app_info->create();
 
-	active_app->set_name(requested_app_info->id);
```
Comment (TomAtkinsonArm) [inline framework/platform/platform.cpp:372]: We might want some method on AppInfo which returns the id if the name is empty?

Something like:

```
active_app->set_name(requested_app_info->human_readable_name());
```

**Review by TomAtkinsonArm — CHANGES_REQUESTED** [review]
LGTM. Suggested a nice to have. If we know for a fact that every sample has a name then this nice to have can be ignored

File: framework/platform/platform.cpp (hunk 1 above)
Comment (asuessenbach) [reply framework/platform/platform.cpp:372]: I think it would be better if the name in AppInfo was mandatory.

Comment (gpx1000) [reply framework/platform/platform.cpp:372]: I tend to agree that AppInfo should be mandatory.  We'll need to ensure all legacy samples are updated for it and it makes its way into acceptance criteria for new sample submissions.  An automated check would be helpful from our CI/CD

**Comment by marty-johnson59** [comment]
This will no longer be an issue once the updated framework (current WIP) is complete.  Closing...can revisit if not fully addressed with the new FW

**Pull request closed without merging**